src/open_options.rs

```diff
@@ -71,9 +74,42 @@ impl OpenOptions {
             truncate: false,
             create: false,
             create_new: false,
+            buffer_size: 0,
+            replication: 0,
+            blocksize: 0,
         }
     }
 
+    /// Sets size of buffer for read/write.
+    ///
+    /// Pass `0` if you want to use the default configured values.
+    ///
+    /// `0` by default.
+    pub fn with_buffer_size(&mut self, buffer_size: c_int) -> &mut Self {
```

Comment:
Hi, how about using rust's own types and convert into `c_int` instead?

Reply:
There is a split about what primitive Rust type it really means in different platforms. "avr" and "msp430" say they are `i16` but the rest of the platforms mean `i32`. I'm not confident if we should choose the strictest route to use `i16` or follow `i32` which is the popular one.

Reply:
My current idea is to use `usize` here and return an error during opening.

Reply:
Please keep in mind that hadoop itself may not be able to run on `avr` or `msp430`.